src/generic_core/globals.ts

```diff
@@ -59,7 +59,9 @@ export var settings :uproxy_core_api.GlobalSettings = {
   activePromoId: null,  // set on promoIdDetected
   enabledExperiments: [],
   shouldHijackDO: true,
-  crypto: true
+  crypto: true,
+  torOn: false,
```

Comment:
I'd recommend putting all Tor config in a sub-object.

The reason why is because it makes it easier to propagate the information the Tor integration needs without having to pass around the entire global settings. The motivation is similar to [_Don’t include primitive types in a top-level request or response proto_](https://docs.google.com/document/d/1boTOKzkRwdr4I1uq2sa1ZAka1mzNadtGhSuuA_w5vSw/preview?hl=en&forcehl=1#heading=h.oyu8cui1gym4).

This will simplify a bunch of places, where you'll have one variable instead of two.
It also avoids some semantic issues, where a field (e.g. torPort) may not make sense if Tor is not enabled.

The settings could look like this when enabled:
````
  tor: { port: 9050 }
````

and like this when disabled:
````
  tor: null
````

You'll need to define a TorSettings class with the port number field.